apk を作成し、github 上にアップロードする

ArimotoSaki opened the pull request:
実行結果（ビルドトリガーだけ実行後に書き換えています）
https://github.com/hs-horie/hello-app/actions/runs/12312565069

master へマージ後されたら、apk が作成されguthub 上へアップロードされるAction です。

念の為、apk 作成前にktlint を実行しています。
また作成されるapk は一旦Debug Build で作成されたのものとなります。

ArimotoSaki committed:
apk を作成し、github 上にアップロードする

ArimotoSaki commented on .github/workflows/gradle.yml:
```diff
@@ -0,0 +1,40 @@
+name: Java CI with Gradle
+
+on:
+  workflow_dispatch:
```
workflow_dispatch は、手動で実行できるようにする設定です。
https://docs.github.com/ja/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/manually-running-a-workflow

動作確認用に追加したのですが、master にマージされないと利用できないようでした。
ただあって困るものでもないと思ったため、残しています。

ArimotoSaki commented on .github/workflows/gradle.yml:
```diff
@@ -0,0 +1,40 @@
+      - name: Build with Gradle
+        run: ./gradlew assembleDebug # releaseBuild にしたい場合は、assembleRelease とする
+      - name: Upload apk
+        uses: actions/upload-artifact@v4
```
actions/upload-artifact を利用する事で、GitHub 上のストレージにアップロードしています。

hs-koguchi commented on .github/workflows/gradle.yml:
```diff
@@ -0,0 +1,40 @@
+
+    steps:
+      - name: Check out
+        uses: actions/checkout@v1
```
@ArimotoSaki 
actions/checkout@v1 は初期のバージョンなので、最新のv4の方がいいと思いました。
(actions/setup-java@v1も同様に)(upload-artifact@ではv4を指定していますし)

ArimotoSaki replied:
それぞれ最新のv4 に設定しました。

ArimotoSaki committed:
Update gradle.yml

checkout とsetup-java を最新のバージョンにする

ArimotoSaki committed:
Update gradle.yml

setup-java の最新が使えなかったので、一つ戻す

ArimotoSaki committed:
Update gradle.yml

distribution として、JetBrains ランタイム（JBR） を指定しました。
これはAndroidStudio と一緒に配布されているJDKであるため、これが最適そうだと思いました。

ArimotoSaki committed:
Update gradle.yml

トリガーをテスト用からmaster 用に変更

ArimotoSaki commented on .github/workflows/gradle.yml:
```diff
@@ -0,0 +1,46 @@
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: "jetbrains"
```
https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions
上記のリンク先にdistribution のサポート一覧が記載されています。

今回はAndroidStudio と合わせてインストールされるJetBrains ランタイム（JBR）を設定しました。
参考: https://developer.android.com/build/jdks?hl=ja
（何が最適かまではちゃんと調査していないです）
